src/interface/guide/components/IntensityBar.tsx

```diff
@@ -0,0 +1,345 @@
+  if (value > 50000) return Math.round(value / 5000) * 5000;
+  return Math.round(value / 1000) * 1000;
+}
+
+interface TargetData {
+  targetName: string;
+  targetID: number;
+  targetInstance?: number;
+  total: number;
+  events: DamageOrHealEvent[];
+}
+
+interface Props {
+  /** The spell or ability being tracked (optional - if omitted, shows all damage/healing) */
+  spell?: Spell;
+  /** Array of per-target damage/healing event data */
+  data: TargetData[];
+  /** Type of intensity chart - 'DPS' or 'HPS'. Default: 'DPS' */
+  chartType?: 'DPS' | 'HPS';
+  /** Custom header override. If not provided, uses "{spell.name} Time Distribution" or "Damage/Healing Time Distribution" */
+  headerOverride?: string;
+  /** Height of the bar in pixels. Default: 60 */
+  height?: number;
+  /** Whether to show percentage labels on segments. Default: true */
+  showLabels?: boolean;
+}
+
+/**
+ * Displays time distribution as a stacked horizontal bar showing what percentage
+ * of time was spent at different intensity tiers based on median-centered thresholds.
+ *
+ * Features:
+ * - Five-tier color gradient (very low to very high intensity)
+ * - Percentage labels on each segment
+ * - Stats showing max DPS/HPS
+ * - Works for both single-spell and overall damage/healing
+ *
+ * @param spell - The spell being tracked (optional - if omitted, shows all damage/healing)
+ * @param data - Array of per-target damage/healing event data
+ * @param chartType - Type of chart: 'DPS' or 'HPS' (default: 'DPS')
+ * @param headerOverride - Custom header text (default: auto-generated from spell/type)
+ * @param height - Height of the bar in pixels (default: 60)
+ * @param showLabels - Whether to show percentage labels on segments (default: true)
+ */
+export default function IntensityBar({
+  spell,
+  data,
+  chartType = 'DPS',
+  headerOverride,
+  height = 45,
+  showLabels = true,
+}: Props) {
+  const info = useInfo();
+  const baseColor = getSpecColor(info?.combatant.spec?.id);
+
+  if (!info || data.length === 0) {
+    return null;
+  }
+
+  const fightStart = info.fightStart;
+  const fightEnd = info.fightEnd;
+
+  // Sample every 1 second to calculate DPS/HPS values
+  const sampleInterval = 1000; // 1 second
+  const samples: number[] = [];
+
+  for (let time = fightStart; time < fightEnd; time += sampleInterval) {
+    const timeStart = time;
+    const timeEnd = time + sampleInterval;
+    let totalAmount = 0;
+
+    data.forEach((target) => {
+      target.events.forEach((event) => {
+        if (event.timestamp >= timeStart && event.timestamp < timeEnd) {
+          totalAmount += event.amount;
+        }
+      });
+    });
+
+    const dps = (totalAmount / sampleInterval) * 1000;
+    if (dps > 0) {
+      samples.push(dps);
+    }
+  }
+
+  if (samples.length === 0) {
+    return null;
+  }
+
+  // Calculate median for threshold centering
+  const sortedSamples = [...samples].sort((a, b) => a - b);
```

Comment:
don't copy samples here
```suggestion
  const sortedSamples = samples.sort((a, b) => a - b);
```